provider.go

```diff
@@ -3,9 +3,10 @@
 //
 // # Updates are not atomic
 //
-// The deSEC API doesn't map 1:1 to the libdns API. The main issue with that is that it's not
-// possible to update records atomically. The implementation here goes to great lengths to avoid
-// interference of multiple concurrent requests, but that only works within a single process.
+// The deSEC API doesn't map 1:1 to the libdns API. The main issue is that libdns works on the
+// record level while the deSEC API works on the RRset level. This makes it impossible to update
+// records atomically (in contrast with RRsets). The implementation here goes to great lengths to
```

Comment:
```suggestion
// RRsets atomically. The implementation here goes to great lengths to
```

Reply:
Text new text says that it is impossible to update records atomically, and it sounds like that's a deSEC limitation (because it's in the README on the deSEC module). In what I suggested, I was trying to avoid that impression.

That said, I think we have a common understanding, and I'll leave it to you to find the right words for this project. I appreciate the effort, but let's not spend too much time on that sentence. I can live with it either way; I mainly opened the #2 because of the TTL thingy.

Reply:
I gave it another go, sorry for the long wait. I hope the current text addresses your concerns.